File: backend/app/schemas/item.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class ItemBase(BaseModel):
    name: str
    category: str
    brand: Optional[str] = None
    season: Optional[str] = None
    fit: Optional[str] = None
    size: Optional[str] = None
    color: Optional[str] = None
    image_key: Optional[str] = None


class ItemCreate(ItemBase):
    pass


class ItemUpdate(BaseModel):
    name: Optional[str] = None
    category: Optional[str] = None
    brand: Optional[str] = None
    season: Optional[str] = None
    fit: Optional[str] = None
    size: Optional[str] = None
    color: Optional[str] = None
    image_key: Optional[str] = None


class ItemRead(ItemBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    image_url: Optional[str] = None
    created_at: datetime

File: mobile/lib/models/item.dart
class Item {
  final int id;
  final String name;
  final String category;
  final String? brand;
  final String? season;
  final String? size;
  final String? imageUrl;

  Item({
    required this.id,
    required this.name,
    required this.category,
    this.brand,
    this.season,
    this.size,
    this.imageUrl,
  });

  factory Item.fromJson(Map<String, dynamic> json) {
    return Item(
      id: json['id'] as int,
      name: json['name'] as String,
      category: json['category'] as String,
      brand: json['brand'] as String?,
      season: json['season'] as String?,
      size: json['size'] as String?,
      imageUrl: json['image_url'] as String?,
    );
  }
}
